src/biz/workspace/publish_dup.rs

```diff
@@ -0,0 +1,941 @@
+      )
+      .await?;
+    Ok(())
+  }
+
+  /// broadcast updates to collab group if exists
+  async fn broadcast_update(&self, oid: &str, encoded_update: Vec<u8>) {
+    tracing::info!("broadcasting update to group: {}", oid);
+    match self.group_manager.get_group(oid).await {
+      Some(group) => {
+        let (collab_message_sender, _collab_message_receiver) = futures::channel::mpsc::channel(1);
+        let (mut message_by_oid_sender, message_by_oid_receiver) =
+          futures::channel::mpsc::channel(1);
+        group
+          .subscribe(
+            &RealtimeUser {
+              uid: self.duplicator_uid,
+              device_id: uuid::Uuid::new_v4().to_string(),
+              connect_at: self.ts_now,
+              session_id: uuid::Uuid::new_v4().to_string(),
+              app_version: "".to_string(),
+            },
+            CollabOrigin::Server,
+            collab_message_sender,
+            message_by_oid_receiver,
+          )
+          .await;
+        let payload = Message::Sync(SyncMessage::Update(encoded_update)).encode_v1();
+        let message = HashMap::from([(
+          oid.to_string(),
+          vec![ClientCollabMessage::ClientUpdateSync {
+            data: UpdateSync {
+              origin: CollabOrigin::Server,
+              object_id: oid.to_string(),
+              msg_id: self.ts_now as u64,
+              payload: payload.into(),
+            },
+          }],
+        )]);
+        match message_by_oid_sender.try_send(message) {
+          Ok(()) => tracing::info!("sent message to group"),
+          Err(err) => tracing::error!("failed to send message to group: {}", err),
+        }
+      },
+      None => tracing::warn!("group not found for oid: {}", oid),
+    }
+  }
+}
+
+async fn get_published_data_for_view_id(
+  txn: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+  view_id: &uuid::Uuid,
+) -> Result<Option<(PublishViewMetaData, Vec<u8>)>, AppError> {
+  match select_published_data_for_view_id(txn, view_id).await? {
+    Some((js_val, blob)) => {
+      let metadata = serde_json::from_value(js_val)?;
+      Ok(Some((metadata, blob)))
+    },
+    None => Ok(None),
+  }
+}
+
+fn view_info_by_view_id(meta: &PublishViewMetaData) -> HashMap<String, PublishViewInfo> {
+  let mut acc = HashMap::new();
+  acc.insert(meta.view.view_id.clone(), meta.view.clone());
+  add_to_view_info(&mut acc, &meta.child_views);
+  add_to_view_info(&mut acc, &meta.ancestor_views);
+  acc
+}
+
+fn add_to_view_info(acc: &mut HashMap<String, PublishViewInfo>, view_infos: &[PublishViewInfo]) {
+  for view_info in view_infos {
+    acc.insert(view_info.view_id.clone(), view_info.clone());
+    if let Some(child_views) = &view_info.child_views {
+      add_to_view_info(acc, child_views);
+    }
+  }
+}
+
+fn collab_from_doc_state(doc_state: Vec<u8>, object_id: &str) -> Result<Collab, AppError> {
+  let collab = Collab::new_with_source(
+    CollabOrigin::Server,
+    object_id,
+    DataSource::DocStateV1(doc_state),
```

Comment:
should reference to the version in `EncodeCollab`
<img width="681" alt="image" src="https://github.com/user-attachments/assets/a8722f13-0b77-47d1-a519-9ff6b2f8d9fa">

Reply:
dont this is possible, as the published blobs are `doc_state` (Bytes) that is assumed to have v1, instead of bytes of `EncodedCollab`